runner/app/live/streamer/process_guardian.py

```diff
@@ -0,0 +1,276 @@
+        await self.update_params(params)
+
+    def send_input(self, frame: InputFrame):
+        if not self.process:
+            raise RuntimeError("Process not running")
+        iss = self.status.input_status
+        previous_input_time = max(iss.last_input_time or 0, self.process.start_time)
+        (iss.last_input_time, iss.fps) = calculate_rolling_fps(
+            iss.fps, previous_input_time
+        )
+
+        self.process.send_input(frame)
+
+    async def recv_output(self) -> OutputFrame | None:
+        if not self.process:
+            raise RuntimeError("Process not running")
+        output = await self.process.recv_output()
+
+        oss = self.status.inference_status
+        previous_output_time = max(oss.last_output_time or 0, self.process.start_time)
+        (oss.last_output_time, oss.fps) = calculate_rolling_fps(
+            oss.fps, previous_output_time
+        )
+        return output
+
+    async def update_params(self, params: dict):
+        if not self.process:
+            raise RuntimeError("Process not running")
+        self.params = params
+        self.process.update_params(params)
+        self.status.update_params(params)
+
+        await self.monitoring_callback(
+            {
+                "type": "params_update",
+                "pipeline": self.pipeline,
+                "params": params,
+                "params_hash": self.status.inference_status.last_params_hash,
+                "update_time": self.status.inference_status.last_params_update_time,
+            }
+        )
+
+    def get_status(self, clear_transient: bool = False) -> PipelineStatus:
+        new_state = self._current_state()
+        if new_state != self.status.state:
+            self.status.state = new_state
+            self.status.last_state_update_time = time.time()
+            logging.info(f"Pipeline state changed to {new_state}")
+        status = self.status.model_copy()
+        if clear_transient:
+            # Clear the large transient fields if requested, but do return them
+            self.status.inference_status.last_params = None
+            self.status.inference_status.last_restart_logs = None
+        return status
+
+    def _current_state(self) -> str:
+        current_time = time.time()
+        input = self.status.input_status
+        last_input_time = input.last_input_time or 0
+        time_since_last_input = current_time - last_input_time
+        if time_since_last_input > 60:
+            if time_since_last_input < 90:
+                # streamer should stop automatically after 60s, so give ourselves a 30s grace period to shutdown
+                logging.info(
+                    f"Detected DEGRADED_INPUT during shutdown: time_since_last_input={time_since_last_input:.1f}s"
+                )
+                return PipelineState.DEGRADED_INPUT
+            return PipelineState.OFFLINE
+        elif time_since_last_input > 2 or input.fps < 15:
```

Comment:
nit: should we make input.fps here configurable using an argument instead? We might end up shipping a dreamshaper workflow that runs right around 15-ish fps.